PWGJE/DataModel/GammaJetAnalysisTree.h

```diff
@@ -0,0 +1,73 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \brief Table definitions for gamma-jet analyses
+///
+/// \author Florian Jonas <florian.jonas@cern.ch>
+
+#ifndef PWGJE_DATAMODEL_GAMMAJETANALYSISTREE_H_
+#define PWGJE_DATAMODEL_GAMMAJETANALYSISTREE_H_
+
+#include "Framework/AnalysisDataModel.h"
+#include "PWGJE/DataModel/EMCALClusters.h"
+#include "PWGJE/Core/JetDerivedDataUtilities.h"
+#include "PWGJE/DataModel/Jet.h"
+
+namespace o2::aod
+{
+namespace gjgamma
+{
+DECLARE_SOA_COLUMN(ColId, colid, int);
+DECLARE_SOA_COLUMN(Energy, energy, float);                             //! cluster energy (GeV)
+DECLARE_SOA_COLUMN(Eta, eta, float);                                   //! cluster pseudorapidity (calculated using vertex)
+DECLARE_SOA_COLUMN(Phi, phi, float);                                   //! cluster azimuthal angle (calculated using vertex)
+DECLARE_SOA_COLUMN(M02, m02, float);                                   //! shower shape long axis
+DECLARE_SOA_COLUMN(M20, m20, float);                                   //! shower shape short axis
+DECLARE_SOA_COLUMN(NCells, nCells, ushort);                            //! number of cells in cluster
+DECLARE_SOA_COLUMN(Time, time, float);                                 //! cluster time (ns)
+DECLARE_SOA_COLUMN(IsExotic, isExotic, bool);                          //! flag to mark cluster as exotic
+DECLARE_SOA_COLUMN(DistanceToBadChannel, distanceToBadChannel, float); //! distance to bad channel
+DECLARE_SOA_COLUMN(NLM, nlm, ushort);                                  //! number of local maxima
+DECLARE_SOA_COLUMN(IsoRaw, isoraw, ushort);                            //! isolation in cone not corrected for Rho
+DECLARE_SOA_COLUMN(PerpConeRho, perpconerho, float);                   //! rho in perpendicular cone
+DECLARE_SOA_COLUMN(TMdeltaPhi, tmdeltaphi, float);                     //! delta phi between cluster and closest match
+DECLARE_SOA_COLUMN(TMdeltaEta, tmdeltaeta, float);                     //! delta eta between cluster and closest match
+DECLARE_SOA_COLUMN(TMtrackP, tmtrackp, float);                         //! track momentum of closest match, -1 if no match found
+} // namespace gjgamma
+DECLARE_SOA_TABLE(GjGammas, "AOD", "GJGAMMA",
+                  gjgamma::ColId, gjgamma::Energy, gjgamma::Eta, gjgamma::Phi, gjgamma::M02, gjgamma::M20, gjgamma::NCells, gjgamma::Time, gjgamma::IsExotic, gjgamma::DistanceToBadChannel, gjgamma::NLM, gjgamma::IsoRaw, gjgamma::PerpConeRho, gjgamma::TMdeltaPhi, gjgamma::TMdeltaEta, gjgamma::TMtrackP)
+namespace gjevent
+{ // TODO add rho
+// DECLARE_SOA_INDEX_COLUMN(JCollision, collision);
+DECLARE_SOA_COLUMN(ColId, colid, int);
```

Comment:
Why do need this CollId?

Reply:
You could instead make it a coloumn that saves the index of the GjEvents table and then fill that (you would need to create a map between this table and jcollisions)